pkg/controller/tiers/tiers_controller.go

```diff
@@ -147,3 +163,49 @@ func (r *ReconcileTiers) Reconcile(ctx context.Context, request reconcile.Reques
 
 	return reconcile.Result{}, nil
 }
+
+func (r *ReconcileTiers) prepareTiersConfig(ctx context.Context, reqLogger logr.Logger) (*tiers.Config /*requeue*/, bool) {
```

Comment:
nit, but it's not very common to inline comments like this in the return values. Instead, I'd recommend using a function-level comment that explains it, like this:

```
// prepareTiersConfig returns a tiers.Config based on cluster state, and a boolean indicating whether or not the controller should requeue the reconcile.
```

It's also possible to use named return values, but I generally advise against those since it's not an especially common pattern. https://go.dev/tour/basics/7

Reply:
In this case though, it seems like we always mark as degraded and schedule a requeue in the same branch. You could just return an error instead of a boolean, which would be a more normal way to propagate errors up to the caller.

The caller would then set degraded on the error, and return it, triggering a requeue anyway.

Reply:
I guess in this case if we return err, we might also need to return the msg we are passing to SetDegraded as well or we have to decide to not care about explaining the details of what happened. is that right? or there is other ways to inject the msg in the error obj?